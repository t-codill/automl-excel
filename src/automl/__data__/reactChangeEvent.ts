// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
