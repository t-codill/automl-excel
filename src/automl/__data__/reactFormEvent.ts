// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
