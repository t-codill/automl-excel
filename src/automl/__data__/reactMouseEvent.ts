// tslint:disable-next-line: no-any
export const reactMouseEvent = {} as unknown as React.MouseEvent<any>;
