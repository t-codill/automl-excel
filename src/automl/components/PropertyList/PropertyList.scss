.propertyList {
    .propertyListProp {
        
        font-weight: 600;
        padding: 15px 10px 10px 10px;
        user-select: text;
    }

    .propertyListValue {
        
        line-height: 2.29;
        font-size: 14px;
        font-style: normal;
        font-stretch: normal;
        letter-spacing: normal;
        text-overflow: ellipsis;
        white-space: nowrap;
        overflow: hidden;
        padding: 10px;
        text-align: right;
        position: relative;
        user-select: text;
    }

    .propertyListValue:hover {

        text-overflow: unset;
        white-space: normal;
    }

    .propertyListProp:hover {
        
        text-overflow: unset;
        white-space: normal;
    }
}