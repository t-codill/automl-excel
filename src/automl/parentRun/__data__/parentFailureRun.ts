import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IParentRunData } from "../ParentRun";

export const parentFailureRun: NotNullableProperties<IParentRunData> = {
    run: {
        runId: "AutoML_000",
        parentRunId: undefined,
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: "",
            primary_metric: "pm",
            errors: "Setup iteration failed: \\n \t\t\t\t\texception: \\n \t\t\t\t\t\tUnforseenConsequence('Invalid subject in the test chamber') \\n"
        }
    },
    experimentName: "Test",
    childRuns: [],
    childRunMetrics: [
        {
            spearman_correlation: 0.877770827248834
        }
    ]
};
