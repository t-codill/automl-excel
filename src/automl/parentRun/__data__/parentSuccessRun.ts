import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IParentRunData } from "../ParentRun";

export const parentSuccessRunAMLSettings = {
  primary_metric: "AUC_weighted",
  data_script: null,
  compute_target: "local",
  n_cross_validations: 3,
  max_cores_per_iteration: 1,
  blacklist_algos: [],
  exit_score: 100,
  name: "my_experiment",
  task_type: "classification"
};
export const parentSuccessRun: NotNullableProperties<IParentRunData> = {
  run: {
    runId: "AutoML_000",
    parentRunId: undefined,
    status: "Completed",
    startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
    endTimeUtc: new Date("2019-02-01T08:30:00.000Z"),
    properties: {
      AMLSettingsJsonString: JSON.stringify(parentSuccessRunAMLSettings)
    },
    tags: {
      iterations: "2"
    },
  },
  experimentName: "Test",
  childRuns: [
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_001",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      experimentId: "exp",
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_001",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_algorithm: "RandomForest",
        run_preprocessor: "MaxAbsScaler",
        score: "0.977770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_002",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_002",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_preprocessor: "MaxAbsScaler",
        score: "0.877770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_003",
      parentRunId: "AutoML_000",
      status: "Canceled",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_003",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        score: "0.877770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      }
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_004",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_004",
      properties: {
        runTemplate: "automl_child",
        pipeline_id: "eda663ea8f36e6c07948c99bba9d1becb40991a5",
        run_template: "automl_child",
        run_algorithm: "MaxAbsScaler",
        score: "0.827770827248834",
        class_labels: "",
        primary_metric: "spearman_correlation",
      },
    },
    {
      createdUtc: new Date("2019-02-01T08:20:00.000Z"),
      runId: "AutoML_005",
      parentRunId: "AutoML_000",
      status: "Completed",
      startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
      endTimeUtc: new Date("2019-02-01T08:25:00.000Z"),
      options: {
        generateDataContainerIdIfNotSpecified: true
      },
      name: "AutoML_005",
    }
  ],
  childRunMetrics: [
    {
      spearman_correlation: 0.877770827248834
    }
  ]
};
