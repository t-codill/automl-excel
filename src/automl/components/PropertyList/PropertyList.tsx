
import { isEmpty, map } from "lodash";
import { CheckboxVisibility, DetailsList, IColumn, MessageBar } from "office-ui-fabric-react";
import * as React from "react";
import { BasicTypes } from "../../common/BasicTypes";
import { IDictionary } from "../../common/IDictionary";

import "./PropertyList.scss";

export class PropertyList extends React.Component<{
    listElements: IDictionary<BasicTypes | React.ReactNode>;
}> {
    private readonly columns: IColumn[] = [
        {
            key: "key",
            name: "key",
            fieldName: "key",
            minWidth: 20,
            maxWidth: 200,
            isResizable: false,
            className: "propertyListProp"
        },
        {
            key: "value",
            name: "value",
            fieldName: "value",
            minWidth: 100,
            isResizable: false,
            className: "propertyListValue"
        }
    ];
    public readonly render = (): React.ReactNode => {
        if (isEmpty(this.props.listElements)) {
            return <MessageBar>No data</MessageBar>;
        }

        return (
            <DetailsList
                className="propertyList"
                columns={this.columns}
                items={this.formatRows()}
                checkboxVisibility={CheckboxVisibility.hidden}
                isHeaderVisible={false} />
        );
    }
    private readonly formatRows = () => {
        return map(this.props.listElements, (value, key) => {
            if (Array.isArray(value)) {
                return { key, value: value.join(", ") };
            }
            return { key, value };
        })
            .filter((item) => item.value);
    }
}
