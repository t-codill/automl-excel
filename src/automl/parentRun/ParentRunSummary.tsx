import { RunHistoryAPIsModels } from "@vienna/runhistory";

import * as React from "react";
import { safeParseJson } from "../common/utils/safeParseJson";
import { PageLoadingSpinner } from "../components/Progress/PageLoadingSpinner";
import { PropertyList } from "../components/PropertyList/PropertyList";

export class ParentRunSummary extends React.Component<{
    run: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDto | undefined;
}> {
    public readonly render = (): React.ReactNode => {
        const run = this.props.run;
        if (!run || !run.properties) {
            return <PageLoadingSpinner />;
        }
        const amlSettingsString = run.properties.AMLSettingsJsonString;
        let parentRunSumData = {};
        if (amlSettingsString) {
            const amlSettings = safeParseJson(amlSettingsString);
            parentRunSumData = {
                "Best Pipeline": amlSettings.best_pipeline,
                "Exit Score": amlSettings.exit_score,
                "Experiment Name": amlSettings.name,
                "Run Id": run.runId,
                "Task Type": amlSettings.is_timeseries ? "forecasting" : amlSettings.task_type,
                "History Name": amlSettings.history_name,
                Iteration: run.tags ? run.tags.iterations : undefined,
                "Preprocessing Steps": amlSettings.preprocessing_steps,
                Status: run.status
            };
        }
        return <PropertyList listElements={parentRunSumData} />;
    }
}
