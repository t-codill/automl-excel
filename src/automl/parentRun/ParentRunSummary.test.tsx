import { shallow } from "enzyme";
import * as React from "react";
import { PropertyList } from "../components/PropertyList/PropertyList";
import { parentFailureRun } from "./__data__/parentFailureRun";
import { parentSuccessRun } from "./__data__/parentSuccessRun";
import { ParentRunSummary } from "./ParentRunSummary";

describe("Parent Run Summary", () => {

    it("should render loading image when no run", () => {
        const tree = shallow(<ParentRunSummary run={undefined} />);
        expect(tree)
            .toMatchInlineSnapshot("<pageLoadingSpinner />");
    });
    it("should render Run Summary table when successful run", () => {
        const tree = shallow(<ParentRunSummary {...parentSuccessRun} />);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should not render iteration tags when undefined", () => {
        const runWithIterationTags = parentSuccessRun;
        if (runWithIterationTags.run) {
            runWithIterationTags.run.tags = undefined;
        }
        const tree = shallow(<ParentRunSummary {...parentSuccessRun} />);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render empty Run Summary table when no aml settings json", () => {
        const tree = shallow(<ParentRunSummary {...parentFailureRun} />);
        expect(tree)
            .toMatchSnapshot();
    });
    it("should show forecasting as task type", () => {
        const tree = shallow(<ParentRunSummary {...parentFailureRun}
            run={
                {
                    ...parentFailureRun.run,
                    properties: {
                        AMLSettingsJsonString: JSON.stringify({
                            is_timeseries: true
                        })
                    }
                }
            }
        />);
        expect(tree.find(PropertyList)
            .prop("listElements")["Task Type"])
            .toBe("forecasting");
    });

});
