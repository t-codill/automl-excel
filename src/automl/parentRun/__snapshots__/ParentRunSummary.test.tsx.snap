// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent Run Summary should not render iteration tags when undefined 1`] = `
<PropertyList
  listElements={
    Object {
      "Best Pipeline": undefined,
      "Exit Score": 100,
      "Experiment Name": "my_experiment",
      "History Name": undefined,
      "Iteration": undefined,
      "Preprocessing Steps": undefined,
      "Run Id": "AutoML_000",
      "Status": "Completed",
      "Task Type": "classification",
    }
  }
/>
`;

exports[`Parent Run Summary should render Run Summary table when successful run 1`] = `
<PropertyList
  listElements={
    Object {
      "Best Pipeline": undefined,
      "Exit Score": 100,
      "Experiment Name": "my_experiment",
      "History Name": undefined,
      "Iteration": "2",
      "Preprocessing Steps": undefined,
      "Run Id": "AutoML_000",
      "Status": "Completed",
      "Task Type": "classification",
    }
  }
/>
`;

exports[`Parent Run Summary should render empty Run Summary table when no aml settings json 1`] = `
<PropertyList
  listElements={Object {}}
/>
`;
